src/states/node.rs

```diff
@@ -1541,22 +1550,41 @@ impl Node {
             let genesis_ver = *genesis_info.our_info.version();
             let consensus_mode = parsec::ConsensusMode::Single;
 
-            if genesis_info.our_info.members().contains(self.id()) {
-                let _ = self.parsec_map.insert(
-                    genesis_ver,
-                    Parsec::from_genesis(full_id, &genesis_info.our_info.members(), consensus_mode),
-                );
+            #[cfg(not(feature = "mock"))]
+            let parsec = if genesis_info.our_info.members().contains(self.id()) {
+                Parsec::from_genesis(full_id, &genesis_info.our_info.members(), consensus_mode)
             } else {
-                let _ = self.parsec_map.insert(
-                    genesis_ver,
+                Parsec::from_existing(
+                    full_id,
+                    &genesis_info.our_info.members(),
+                    &genesis_info.latest_info.members(),
+                    consensus_mode,
+                )
+            };
+
+            #[cfg(feature = "mock")]
+            let parsec = {
+                let section_info = genesis_info.our_info.prefix().with_version(genesis_ver);
```

Comment:
ah can you make this not rely on `VersionedPrefix` here pls. That type is just lingering code left behind until the routing_table state gets derived from the Chain itself. If this just took a Digest(Prefix+ver), we're prolly better off here or would have to make that change when we remove that type anyways.

Also maybe name the variable ver_pfx or something pls. `section_info` is an actual type(`SectionInfo`) used in the Chain and might be confusing this called the same.